cfme/tests/infrastructure/test_provisioning_rest.py

```diff
@@ -73,6 +74,20 @@ def provision_data(appliance, provider, small_template_modscope):
     return get_provision_data(appliance.rest_api, provider, small_template_modscope.name)
 
 
+def clean_vm(provider, vm_name):
+    if not provider.mgmt.is_vm_stopped(vm_name):
```

Comment:
https://github.com/lkhomenk/integration_tests/blob/1bba00780339341dc1c259f9d13db0119d893686/cfme/common/vm.py#L611 - we already have this. It might be useful

Reply:
I'd agree with @lkhomenk here unless there's something preventing use of `delete_from_provider()`  please use that (or modify it to support your use case).